src/component/segments/SegmentList/SegmentList.tsx

```diff
@@ -0,0 +1,144 @@
+import { useStyles } from './SegmentList.styles';
+import { useSegments } from 'hooks/api/getters/useSegments/useSegments';
+import { SegmentDeleteConfirm } from '../SegmentDeleteConfirm/SegmentDeleteConfirm';
+import { useSegmentsApi } from 'hooks/api/actions/useSegmentsApi/useSegmentsApi';
+import useToast from 'hooks/useToast';
+import { formatUnknownError } from 'utils/format-unknown-error';
+
+export const SegmentsList = () => {
+    const { hasAccess } = useContext(AccessContext);
+    const { segments, refetchSegments } = useSegments();
+    const { deleteSegment } = useSegmentsApi();
+    const { page, pages, nextPage, prevPage, setPageIndex, pageIndex } =
+        usePagination(segments, 10);
+    const [currentSegment, setCurrentSegment] = useState<ISegment | null>(null);
+    const [delDialog, setDelDialog] = useState(false);
+    const [confirmName, setConfirmName] = useState('');
+    const { setToastData, setToastApiError } = useToast();
+
+    const styles = useStyles();
+
+    const onDeleteSegment = async () => {
+        if (!currentSegment?.id) return;
+        try {
+            await deleteSegment(currentSegment?.id);
+            refetchSegments();
+            setToastData({
+                type: 'success',
+                title: 'Successfully deleted segment',
+                text: 'Your segment is now deleted',
+            });
+        } catch (error: unknown) {
+            setToastApiError(formatUnknownError(error));
+        }
+        setDelDialog(false);
+        setConfirmName('');
+    };
+
+    const renderSegments = () => {
+        return page.map((segment: ISegment) => {
+            return (
+                <SegmentListItem
+                    key={segment.id}
+                    id={segment.id}
+                    name={segment.name}
+                    description={segment.description}
+                    createdAt={segment.createdAt}
+                    createdBy={segment.createdBy}
+                    //@ts-expect-error
+                    setCurrentSegment={setCurrentSegment}
+                    setDelDialog={setDelDialog}
+                />
+            );
+        });
+    };
+
+    const renderNoSegments = () => {
+        return (
+            <div className={styles.container}>
+                <Typography className={styles.title}>
+                    There are no segments created yet.
+                </Typography>
+                <p className={styles.subtitle}>
+                    Segment makes it easy for you to define who should be
+                    exposed to your feature. The segment is often a collection
+                    of constraints and can be reused.
+                </p>
+                <PermissionButton
+                    onClick={() => {}}
+                    variant="outlined"
+                    color="secondary"
+                    className={styles.paramButton}
+                    permission={ADMIN}
```

Comment:
There are new permissions for `CREATE/UPDATE/DELETE_SEGMENT` that we should use here.